homeassistant/components/sensor/filter.py

```diff
@@ -0,0 +1,267 @@
+"""
+Allows the creation of a sensor that filters state property.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.filter/
+"""
+import asyncio
+import logging
+import statistics
+from collections import deque
+
+import voluptuous as vol
+
+from homeassistant.util import slugify
+from homeassistant.core import callback
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_NAME, CONF_ENTITY_ID, ATTR_UNIT_OF_MEASUREMENT, ATTR_ENTITY_ID)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_state_change
+
+_LOGGER = logging.getLogger(__name__)
+
+FILTER_NAME_LOWPASS = 'lowpass'
+FILTER_NAME_OUTLIER = 'outlier'
+
+CONF_FILTERS = 'filters'
+CONF_FILTER_NAME = 'filter'
+CONF_FILTER_WINDOW_SIZE = 'window_size'
+CONF_FILTER_PRECISION = 'precision'
+CONF_FILTER_RADIUS = 'radius'
+CONF_FILTER_TIME_CONSTANT = 'time_constant'
+
+DEFAULT_FILTER_RADIUS = 2.0
+DEFAULT_FILTER_TIME_CONSTANT = 10
+
+NAME_TEMPLATE = "{} filter"
+ICON = 'mdi: chart-line-variant'
+
+FILTER_SCHEMA = vol.Schema({
+    vol.Optional(CONF_FILTER_WINDOW_SIZE): vol.Coerce(int),
+    vol.Optional(CONF_FILTER_PRECISION): vol.Coerce(int),
+})
+
+FILTER_OUTLIER_SCHEMA = FILTER_SCHEMA.extend({
+    vol.Required(CONF_FILTER_NAME): FILTER_NAME_OUTLIER,
+    vol.Optional(CONF_FILTER_RADIUS): vol.Coerce(float),
+})
+
+FILTER_LOWPASS_SCHEMA = FILTER_SCHEMA.extend({
+    vol.Required(CONF_FILTER_NAME): FILTER_NAME_LOWPASS,
+    vol.Optional(CONF_FILTER_TIME_CONSTANT): vol.Coerce(int),
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(ATTR_ENTITY_ID): cv.entity_id,
```

Comment:
I think this can be wrong interpreted from user. Maybe 'entity', 'state_from' is better?

Reply:
I would be for `entity:` since the plural form of that (`entities:`) is already used for `group:` and the upcoming group platforms.

Reply:
I used entity_id (singular) since that is what is used in template_sensor, min_max, history_stats (and I guess couple of other...)

So we either change all the others, or this is the common interpretation by users already...